python/ffcc/io.py

```diff
@@ -14,145 +14,153 @@
-        tf.logging.error('Unable to load %s', sub_dir)
+  return rgb_illuminant, weight, rgb, extended_feature
 
-  return rgb_illuminant, weight, frames
 
-
-def encode_to_tfrecord(data, output_dir):
-  """Encodes a data item to TFRecord format.
+def get_training_eval_sets(files, cvfolds, test_fold):
+  """Retrieve train and evaluation sets for the current testing fold .
 
   Args:
-    data: A dictionary maps to the FFCC data bundle.
-    output_dir: The output folder. The record will be saved as <name>.tfrecord.
+    files: A list of filenames that should be synched with the cvfolds.
+    cvfolds: A list of corresponding fold number of each filename in 'files'
+    test_fold: Number of testing fold (should be in the range [1-3]).
+
+  Returns:
+    train_set and eval_set, lists of training and evaluation
+      filenames, respectively.
   """
+  assert len(files) == len(cvfolds)
+  assert test_fold in range(1,4)
+  train_folds = list({1, 2, 3} - {test_fold})
+  train_set = []
+  for train_fold_i in train_folds:
+    train_set = np.append(train_set,
+                          np.array(files).transpose()[
+                            (cvfolds == train_fold_i)])
+  train_set = train_set.tolist()
 
-  def _int64_feature(value):
-    """Converts a list of int64 into TF Feature."""
-    return tf.train.Feature(
-        int64_list=tf.train.Int64List(value=np.atleast_1d(value)))
-
-  def _float_feature(value):
-    """Converts a list of fp32 into TF Feature."""
-    return tf.train.Feature(
-        float_list=tf.train.FloatList(value=np.atleast_1d(value)))
-
-  def _bytes_feature(value):
-    """Converts a list of bytes into TF Feature."""
-    # Note: tf.train.BytesList does not work well with np.asleast_1d. If input
-    # value is all 0s, then tf.train.BytesList would truncate the result from
-    # np.asleast_1d(value) as a emptry string.
-    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))
-
-  name = data['name']
-  rgb = data['rgb']
-  extended_feature = data['extended_feature']
-  illuminant = data['illuminant']
-  weight = data['weight']
-
-  filename = os.path.join(output_dir, name + '.tfrecord')
-  writer = tf.python_io.TFRecordWriter(filename)
-
-  height, width, channel = rgb.shape
-  example = tf.train.Example(
-      features=tf.train.Features(
-          feature={
-              'name': _bytes_feature(name.encode('utf-8')),
-              'height': _int64_feature(height),
-              'width': _int64_feature(width),
-              'channel': _int64_feature(channel),
-              'rgb': _bytes_feature(rgb.tobytes()),
-              'extended_feature': _float_feature(extended_feature),
-              'illuminant': _float_feature(illuminant),
-              'weight': _float_feature(weight)
-          }))
-  writer.write(example.SerializeToString())
-  writer.close()
-
-
-def decode_from_tfrecord_proto(example_proto):
-  """Decodes TFRecord protobuf.
+  eval_set = np.array(files)[(cvfolds == test_fold)]
```

Comment:
I think the python list comprehension is slightly easier to read:

eval_set = [f for i, f in enumerate(files) if cvfolds[i] == test_fold]